week2/homework/src/actions.js

```diff
@@ -0,0 +1,78 @@
+const fs = require('fs');
+const util = require('util');
+
+const readFile = util.promisify(fs.readFile);
+
+const actions = {
+  add: element => {
+    readFile('todo.txt', 'utf-8').then(data => {
+      if (data === '') {
+        fs.appendFile('todo.txt', element, err => {
+          if (err) throw err;
+          console.log('an item has been added ...');
+        });
+      } else {
+        fs.appendFile('todo.txt', `\n${element}`, err => {
```

Comment:
you're repeating the same function twice, that's a sign that something can be extracted.
you can either write a function that does all that and you pass it `element` or `element/n`
```js
const saveTodoItem = element => {
  fs.appendFile('todo.txt', element, err => {
    if (err) throw err;
    console.log('an item has been added ...');
  });
}
```

and you call it like 
```js
if(data === '') saveTodoItem(element)
else saveTodoItem(`\n${element}`)
```

OR you can simply create an intermediate variable to append
```js
const line = data === '' ? element : `\n${element}`;
```

Reply:
:clap: